packages/formstr-app/src/containers/Dashboard/FormCards/MyForms.tsx

```diff
@@ -5,59 +5,144 @@ import { getDefaultRelays } from "../../../nostr/common";
-    let myForms = await pool.querySync(getDefaultRelays(), myFormsFilter);
-    console.log("Final my forms", myForms);
-    setFormEvents(myForms);
-    setRefreshing(false);
-    pool.close(getDefaultRelays());
-  };
+  const fetchFormEvents = async (forms: Tag[], existingPool?: SimplePool) => {
+    try {
+      const dTags = forms.map((f) => f[1].split(":")[1]);
+      const pubkeys = forms.map((f) => f[1].split(":")[0]);
+      
+      let myFormsFilter = {
+        kinds: [30168],
+        "#d": dTags,
+        authors: pubkeys,
+      };
+      console.log("myFormsFilter", myFormsFilter);
+      
+      // Use the existing pool if provided, otherwise create new one
+      const pool = existingPool || new SimplePool();
+      let myForms = await pool.querySync(getDefaultRelays(), myFormsFilter);
+      setFormEvents(myForms);
+      
+      // Only close if we created a new pool
+      if (!existingPool) {
+        pool.close(getDefaultRelays());
+      }
+    } catch (error) {
+      console.error('Error fetching form events:', error);
+    } finally {
+      setRefreshing(false);
+    }
+};
 
-  useEffect(() => {
-    const fetchMyForms = async () => {
-      if (!userPub) return;
-      setRefreshing(true);
+const fetchMyForms = async (existingPool?: SimplePool) => {
+    if (!userPub) return;
+    
+    setRefreshing(true);
+    const pool = existingPool || new SimplePool();
+    
+    try {
       let existingListFilter = {
         kinds: [14083],
         authors: [userPub],
       };
-      let pool = new SimplePool();
+      
       let myFormsList = await pool.get(getDefaultRelays(), existingListFilter);
-      console.log("myFormList", myFormsList);
+      
       if (!myFormsList) {
         setRefreshing(false);
         return;
       }
+      
       let forms = await window.nostr.nip44.decrypt(
         userPub,
         myFormsList.content
       );
-      console.log("myFormList", forms);
-      fetchFormEvents(JSON.parse(forms), pool);
+      
+      await fetchFormEvents(JSON.parse(forms), pool);
+    } catch (error) {
+      console.error('Error fetching forms:', error);
+      setRefreshing(false);
+    } finally {
+      if (!existingPool) {
+        pool.close(getDefaultRelays());
+      }
+    }
+};
+
+const handleFormDeleted = async (formId: string) => {
+  if (!userPub) return;
+  setRefreshing(true);
+  const pool = new SimplePool();
+  
+  try {
+    const existingListFilter = {
+      kinds: [14083],
+      authors: [userPub],
     };
-    if (userPub) fetchMyForms();
-  }, [userPub]);
+    
+    const myFormsList = await pool.get(getDefaultRelays(), existingListFilter);
+    
+    if (!myFormsList) {
+      console.error('No forms list found');
+      return;
+    }
+
+    const forms = JSON.parse(await window.nostr.nip44.decrypt(
+      userPub,
+      myFormsList.content
+    ));
+
+    const updatedForms = forms.filter((f: Tag) => {
+      const [_, formKey] = f[1].split(":");
```

Comment:
should we match the whole pubkey + form Id  string? in case a person has the same formIds for different pubkeys

Reply:
?